app/src/gui/fields/TakePhoto.tsx

```diff
@@ -220,38 +200,53 @@ const FAIMSImageList = (props: ImageListProps) => {
     </ImageGalleryList>
   );
 };
-interface State {
-  open: boolean;
-  photopath: string | null;
-  images: Array<any>;
-}
-export class TakePhoto extends React.Component<
+
+export const TakePhoto: React.FC<
   FieldProps &
     Props &
     ButtonProps & {
       ValueTextProps: React.HTMLAttributes<HTMLSpanElement>;
       ErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
       NoErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
-    },
-  State
-> {
-  constructor(
-    props: FieldProps &
-      Props &
-      ButtonProps & {
-        ValueTextProps: React.HTMLAttributes<HTMLSpanElement>;
-        ErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
-        NoErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
+    }
+> = props => {
+  const [open, setOpen] = React.useState(false);
+  const [photoPath, setPhotoPath] = React.useState<string | null>(null);
+  const [noPermission, setNoPermission] = React.useState<boolean>(false);
+  const [images, setImages] = React.useState<Array<any>>(
+    props.form.values[props.field.name] ?? []
+  );
+
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  React.useEffect(() => {
+    const value = props.form.values[props.field.name];
+    if (value !== null && value !== undefined) {
+      setImages(value);
+    }
+  }, [props.form.values[props.field.name]]);
```

Comment:
Bit of an anti-pattern - if the only purpose of the images state is to remain synchronised with the props.form.values value, just declare a const of that value and use it.

Reply:
Dropped the variable all together, just use the prop value directly.